### PR TITLE
Assignment 1

### PR DESCRIPTION
**Pull request by JerettLatimer**

### CODE REVIEW

File: SecretSanta/src/SecretSanta.Business/Gift.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+namespace SecretSanta.Business
+{
+    public class Gift
+    {
+        public int Id { get; } 
+        public string Title { get; set; }
+        public string Description { get; set; }
+        public string Url { get; set; }
+        public User User { get; set; }
```
Comment (mcstudor) [inline SecretSanta/src/SecretSanta.Business/Gift.cs]: These set methods need to check for null inputs

File: SecretSanta/src/SecretSanta.Business/User.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+using System.Text;
+
+namespace SecretSanta.Business
+{
+    public class User
+    {
+        public int Id { get; }
+        public string FirstName { get; set; }
+        public string LastName { get; set; }
+        public List<Gift> Gifts { get; }
```
Comment (mcstudor) [inline SecretSanta/src/SecretSanta.Business/User.cs]: These set methods need to check for null inputs

**Comment by mcstudor** [comment]
# Assignment

1. Turn on nullability in the **SecretSanta.API** project ✔
2. Configure nullability in **SecretSanta.API** project to be conditional based Nullability not already being set (pending Thursday lecture). ❌
3. Add the following classes to the **SecretSanta.Business** project with the following the associated properties:
   - **`Gift`** ✔
     - `int Id` as read-only ✔
     - `string Title` ✔
     - `string Description` ✔
     - `string Url` ✔
     - `User User` ✔

   - **`User`** ✔
     - `int Id` as read-only ✔
     - `string FirstName` ✔
     - `string LastName` ✔
     - <*a collection of*> `Gifts` ✔

4. Add non-default constructors for `Gift` and `User`. ✔
5. Add unit tests to all projects except **SecretSanta.Web** and fully unit test the new classes in **SecretSanta.Business**(pending Thursday lecture). ✔
6. Refactor nullability setting into solution level **props** and **targets** files (pending Thursday lecture). ✔
7. Add the following list of code analysis assemblies and appropriately handle all warnings: `IntelliTectAnalyzer.dll`,`Microsoft.NetCore.Analyzers.dll`,`Microsoft.CodeQuality.Analyzers.dll`, and `Microsoft.NetCore.Analyzers.dll`. (Adding the IntelliTect one will automatically add the others since they are dependencies on the IntelliTect one.)
8. Refactor out a solution level global suppression file for disabling code analysis warnings across all projects (to do this add a reference to the same suppression file(s) from all projects). ✔
9. Configure Azure DevOps build for continuous integration to compile and run all unit tests ✔
10. If any updates occur in **Assignment1** prior to your PR, rebase onto **Assignment1**.  Finally, and just prior to submitting your PR, rebase from master (which has additional commits added after **Assessment1** was created such as the addition of the cSpell.json in commit #7b106a6). ✔

## Extra Credit

The following are options for extra credit (you don't need to do them all):

- Use reflection to test all properties on all classes in **SecretSanta.Business**. ❌
- Configure tests to run in parallel both locally and in Azure DevOps and be sure the both the test and SUT code is appropriately thread safe for the tests to pass reliably. ❌

**Comment by mcstudor** [comment]
Overall looked good!

**Comment by JerettLatimer** [comment]
Pipeline Link: https://dev.azure.com/jlatimer1/CSCD379-2020-Winter/_build?definitionId=1&_a=summary

**Pull request closed without merging**